### PR TITLE
Add policy addon

### PR DESCRIPTION
**Pull request by zhiweiyin318**
Signed-off-by: Zhiwei Yin <zyin@redhat.com>

### CODE REVIEW

**Comment by mikeshng** [comment]
/assign @mprahl @dhaiducek @JustinKuli 

Please help take a look. Thanks!

File: controlplane/hack/deploy/addon/policy/manager/addon-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+          env:
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: metadata.name
+            - name: CONFIG_POLICY_CONTROLLER_IMAGE
+              value: quay.io/open-cluster-management/config-policy-controller:latest
+            - name: KUBE_RBAC_PROXY_IMAGE
+              value: registry.redhat.io/openshift4/ose-kube-rbac-proxy:v4.10
```
Comment (morvencao) [inline controlplane/hack/deploy/addon/policy/manager/addon-deployment.yaml:75]: need credential to pull this image?

Comment (qiujian16) [reply controlplane/hack/deploy/addon/policy/manager/addon-deployment.yaml:75]: why need rbac proxy?

Comment (zhiweiyin318) [reply controlplane/hack/deploy/addon/policy/manager/addon-deployment.yaml:75]: the config-policy-controller agent pod will create proxy container when the managed cluster is OCP. 
https://github.com/open-cluster-management-io/governance-policy-addon-controller/blob/353c6ae127dc3b9b413438749b7f970f6559a8c1/pkg/addon/configpolicy/manifests/managedclusterchart/templates/deployment.yaml#L33

**Review by morvencao — COMMENTED** [review]
/lgtm

File: controlplane/hack/deploy/addon/policy/manager/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+- addon-deployment.yaml
+- propagator-deployment.yaml
+- propagator-metrics-service.yaml
+- role.yaml
+- rolebinding.yaml
+- serviceaccount.yaml
+
+images:
+- name: quay.io/open-cluster-management/governance-policy-addon-controller:latest
+  newName: quay.io/zhiweiyin/governance-policy-addon-controller
```
Comment (mprahl) [inline controlplane/hack/deploy/addon/policy/manager/kustomization.yaml:14]: Is this intentional?

File: controlplane/hack/deploy/addon/policy/hub/clustermanagementaddon/clustermanagementaddon.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+apiVersion: addon.open-cluster-management.io/v1alpha1
```
Comment (mprahl) [inline controlplane/hack/deploy/addon/policy/hub/clustermanagementaddon/clustermanagementaddon.yaml:1]: Do you also have the `governance-policy-framework` `ClusterManagementAddOn` manifest?

Comment (zhiweiyin318) [reply controlplane/hack/deploy/addon/policy/hub/clustermanagementaddon/clustermanagementaddon.yaml:1]: we only add config-policy-addon for test at the first stage.

**Pull request merged**